### PR TITLE
Test cleanup + instability research

### PR DESCRIPTION
**Pull request by rosstuck**
Was cleaning up a couple things and found a couple of quirky issues.

In #53, the phpunit config was changed to always run in process isolation, so I went to remove the remaining tags, silence some warnings, etc. You can see that part below.

However, I found two issues which I'll mention in review comments below:

### CODE REVIEW

File: tests/Security/Voter/HandleCommandVoterTest.php [hunk 1]
```diff
@@ -50,22 +50,62 @@ public function testVote($attribute, $subject, $decision, $mapping, $expected)
...
+                new FakeCommand,
+                true,
+                [],
+                VoterInterface::ACCESS_ABSTAIN
+            ],
+            'deny access if decision manager returns false, even if the command is in the mapping' => [
+                'handle',
+                new FakeCommand,
+                false,
+                [FakeCommand::class => ['ROLE_USER']],
```
Comment (rosstuck) [inline tests/Security/Voter/HandleCommandVoterTest.php]: First off, these mappings here look legit for the test fixtures but they don't actually do anything: the AccessDecisionManager is mocked so the test passes without caring that this here.

So, we should either create a better mock or remove these from the test. That could cleanup the number of test cases we have here and we can move more of this logic to the new integration tests which actually work with the mappings.

File: tests/Security/Voter/HandleCommandVoterTest.php [hunk 2]
```diff
@@ -50,22 +50,62 @@ public function testVote($attribute, $subject, $decision, $mapping, $expected)
...
+                VoterInterface::ACCESS_DENIED
+            ],
+            'grant access if decision manager returns true and the command is in the mapping' => [
+                'handle',
+                new FakeCommand,
+                true,
+                [FakeCommand::class => ['ROLE_USER']],
```
Comment (rosstuck) [inline tests/Security/Voter/HandleCommandVoterTest.php]: There's another, different issue with this test. Although it's near identical to the previous one, removing the mapping here for an empty array should still pass (remember, the decision manager is mocked). However, emptying the array crashes the test with the following:

```
1) League\Tactician\Bundle\Tests\Security\Voter\HandleCommandVoterTest::testVote with data set "grant access if decision manager returns true and the command is in the mapping" ('handle', League\Tactician\Bundle\Tests\Fake\FakeCommand Object (), true, array(), 1)
PHPUnit\Framework\Exception: PHP Fatal error:  Uncaught BadMethodCallException: Method Mockery_1_Symfony_Component_Security_Core_Authentication_Token_TokenInterface::serialize() does not exist on this mock object in [...]/vendor/mockery/mockery/library/Mockery/Loader/EvalLoader.php(16) : eval()'d code:709
Stack trace:
#0 [...]/vendor/mockery/mockery/library/Mockery/Loader/EvalLoader.php(16) : eval()'d code(839): Mockery_1_Symfony_Component_Security_Core_Authentication_Token_TokenInterface->_mockery_handleMethodCall('serialize', Array)
#1 [internal function]: Mockery_1_Symfony_Component_Security_Core_Authentication_Token_TokenInterface->serialize()
#2 -(270): serialize(Array)
#3 -(386): __phpunit_run_isolated_test()
#4 {main}
```

This appears to be related to the processIsolation flag being flipped on everywhere and, indeed, reverting the phpunit.xml.dist flag back allows this test to pass.

So we might want to switch back to making processIsolation opt-in and again, revisit this test setup.

File: tests/Security/Voter/HandleCommandVoterTest.php [hunk 3]
```diff
@@ -50,22 +50,62 @@ public function testVote($attribute, $subject, $decision, $mapping, $expected)
...
+                VoterInterface::ACCESS_GRANTED
+            ],
+            'deny access if the command is not in the mapping (i.e. a default deny access case)' => [
```
Comment (rosstuck) [inline tests/Security/Voter/HandleCommandVoterTest.php]: In case you're wondering, this one acts like the first use-case above: it passes without crashing when the mapping is removed.

File: tests/Security/Voter/HandleCommandVoterTest.php (hunk 1 above)
Comment (RonRademaker) [inline tests/Security/Voter/HandleCommandVoterTest.php]: I don't think this test is entirely incorrect, but perhaps it's a bit confusing. This test tests if the decision manager is actually used if the command is in the mapping. If it's not in the mapping, the default access will be `false` (https://github.com/thephpleague/tactician-bundle/blob/master/src/Security/Voter/HandleCommandVoter.php#L72), because of the mapping in this test, we can test this other line (https://github.com/thephpleague/tactician-bundle/blob/master/src/Security/Voter/HandleCommandVoter.php#L69).

File: tests/Security/Voter/HandleCommandVoterTest.php (hunk 2 above)
Comment (RonRademaker) [inline tests/Security/Voter/HandleCommandVoterTest.php]: If you remove the mapping, the command will never make it to the decision manager. 

About the isolation flip, I flipped it because I was getting errors when booting the kernel multiple times in the integration tests. 

```
PHP Fatal error:  Cannot declare class testappTestDebugProjectContainer, because the name is already in use in /private/var/folders/v9/wt4_2v_55hxf__cwmclw2x0m0000gn/T/tactician-bundle/tactician-bundle594b73b3a08281.73196175/testappTestDebugProjectContainer.php on line 1738
```

File: tests/Security/Voter/HandleCommandVoterTest.php (hunk 3 above)
Comment (RonRademaker) [inline tests/Security/Voter/HandleCommandVoterTest.php]: Indeed we can remove the mapping here safely.

File: src/Security/Voter/HandleCommandVoter.php [hunk 4]
```diff
@@ -25,7 +25,7 @@ class HandleCommandVoter extends Voter
      *
      * @var array
      */
-    private $commandRoleMapping;
+    private $commandRoleMapping = [];
```
Comment (RonRademaker) [inline src/Security/Voter/HandleCommandVoter.php]: Why did you add this? It's constructor argument so it's always initialized there or am I missing something?

Comment (rosstuck) [inline src/Security/Voter/HandleCommandVoter.php]: Pure habit, sorry. 😅

Comment (RonRademaker) [inline src/Security/Voter/HandleCommandVoter.php]: NP, it's a habit that won't hurt anyone 👍

**Comment by rosstuck** [comment]
Got to stop for today but I found that mocking the AccessDecisionManager directly suddenly made a lot of the Voter unit tests clearer in what they do. I also "fixed" some of the process isolation related crashes, added another integration test case for the security component and did some research into other defaults that could cause issues we don't want. For example, what happens to our default security settings if someone changes the AccessDecisionManager's default strategy or consensus settings. So far it looks alright though it might need some extra docs.

I think these are fairly small changes though but they improve the cases we're covering a bit. Lemme know what y'all think and I'll pick it up again tomorrow. :)
